### PR TITLE
Replace AUTHORITY_KEY with ClientStream.setAuthority

### PR DESCRIPTION
**Pull request by ejona86**

### CODE REVIEW

**Comment by ejona86** [comment]
@carl-mastrangelo, friendly ping.

File: netty/src/main/java/io/grpc/netty/NettyClientStream.java [hunk 1]
```diff
@@ -52,30 +55,62 @@
...
   }
 
   @Override
   public void start(ClientStreamListener listener) {
     super.start(listener);
-    startCallback.run();
+
+    // Convert the headers into Netty HTTP/2 headers.
+    AsciiString defaultPath = new AsciiString("/" + method.getFullMethodName());
+    Http2Headers http2Headers = Utils.convertClientHeaders(
```
Comment (carl-mastrangelo) [inline netty/src/main/java/io/grpc/netty/NettyClientStream.java]: nit: break at highest semantic level; in this case the =.

File: okhttp/src/main/java/io/grpc/okhttp/OkHttpClientStream.java [hunk 2]
```diff
@@ -125,9 +127,20 @@ public Integer id() {
...
+  public void setAuthority(String authority) {
+    checkState(listener() == null, "must be call before start");
+    this.authority = checkNotNull(authority, "authority");
+  }
+
+  @Override
   public void start(ClientStreamListener listener) {
     super.start(listener);
-    startCallback.run();
+    String defaultPath = "/" + method.getFullMethodName();
```
Comment (carl-mastrangelo) [inline okhttp/src/main/java/io/grpc/okhttp/OkHttpClientStream.java]: maybe clear `headers` here?

**Comment by carl-mastrangelo** [comment]
LGTM

File: okhttp/src/main/java/io/grpc/okhttp/OkHttpClientStream.java (hunk 2 above)
Comment (ejona86) [inline okhttp/src/main/java/io/grpc/okhttp/OkHttpClientStream.java]: Done.

File: netty/src/main/java/io/grpc/netty/NettyClientStream.java (hunk 1 above)
Comment (ejona86) [inline netty/src/main/java/io/grpc/netty/NettyClientStream.java]: Done

**Pull request merged**